geode-docs/developing/outside_data_sources/implementing_data_loaders.html.md.erb

```diff
@@ -19,70 +19,176 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-To program a data loader and configure your region to use it:
+To use a data loader:
 
-1. Program your loader.
+1. Implement the `org.apache.geode.cache.CacheLoader` interface.
 
-2. Install your loader in each member region where you need it.
+2. Configure and deploy the implementation.
 
-## <a id="implementing_data_loaders__section_88076AF5EC184FE88AAF4C806A0CA9DF" class="no-quick-link"></a>Program your loader
-To program your loader:
+## <a id="implementing_data_loaders__section_88076AF5EC184FE88AAF4C806A0CA9DF" class="no-quick-link"></a>Implement the CacheLoader Interface
 
-1.  Implement `org.apache.geode.cache.CacheLoader`.
+For a get operation, if the key is not in the cache,
+the thread invokes the `CacheLoader.load` method.
+Implement `load` to return the value for the key,
+which will be placed into the region in addition to being
+returned to the caller.
 
-2.  If you want to declare the loader in your `cache.xml`, implement the `org.apache.geode.cache.Declarable` interface as well.
+`org.apache.geode.cache.CacheLoader` inherits from `Declarable`,
+so implement the `Declarable.initialize` method if
+the initialization of an object instantiated from the class
+implementing the `CacheLoader` interface uses an argument.
+Also implement the `Declarable.initialize` method if
+the argument is defined in a `cache.xml` file.
+Do not define the `Declarable.init()` method; it is deprecated.
+Specify the argument in the your configuration.
+This will either be in the `cache.xml` file or in a gfsh
+`create region` or `alter region` command.
```

Comment:
Specify the required arguments either in your `cache.xml` file or in a gfsh
`create region` or `alter region` command.

Do not define the `Declarable.init()` method; it is deprecated.